pkg/migrate/helpers.go

```diff
@@ -0,0 +1,454 @@
+			"nodeSelector":                  sts.Spec.Template.Spec.NodeSelector,
+			"tolerations":                   sts.Spec.Template.Spec.Tolerations,
+			"terminationGracePeriodSeconds": sts.Spec.Template.Spec.TerminationGracePeriodSeconds,
+			"loggingConfigMapName":          input.loggingConfigMap,
+			"env":                           sts.Spec.Template.Spec.Containers[0].Env,
+		},
+	}
+}
+
+func generateParsedMigrationInput(
+	ctx context.Context,
+	clientset kubernetes.Interface,
+	sts *appsv1.StatefulSet) (parsedMigrationInput, error) {
+	var startCmd string
+	var envVars = make(map[string]string)
+	var parsedInput = parsedMigrationInput{
+		tlsEnabled: true,
+	}
+
+	// In the public Helm chart, logging configuration is provided as a secret to the StatefulSet.
+	// However, in the Cockroach Enterprise Operator, it is supplied as a ConfigMap.
+	for _, vol := range sts.Spec.Template.Spec.Volumes {
+		if vol.Name == "log-config" {
+			if vol.Secret != nil {
+				parsedInput.loggingConfigMap = vol.Secret.SecretName
+				if err := ConvertSecretToConfigMap(ctx, clientset, sts.Namespace, parsedInput.loggingConfigMap); err != nil {
+					return parsedInput, err
+				}
+			}
+		}
+	}
+
+	for _, c := range sts.Spec.Template.Spec.Containers {
+		if c.Name == "db" {
+			startCmd = c.Args[2]
+			for i := range c.Env {
+				envVars[c.Env[i].Name] = c.Env[i].Value
+			}
+		}
+	}
+
+	if err := extractJoinStringAndFlags(&parsedInput, strings.Fields(startCmd), envVars); err != nil {
+		return parsedInput, err
+	}
+
+	return parsedInput, nil
+}
+
+// extractJoinStringAndFlags parses the command arguments, extracts the --join string, and replaces env variables.
+func extractJoinStringAndFlags(
+	parsedInput *parsedMigrationInput,
+	args []string,
+	envVars map[string]string) error {
+
+	flags := make(map[string]string)
+	// Regular expression to match flags (e.g., --advertise-host=something)
+	flagRegex := regexp.MustCompile(`--([\w-]+)=(.*)`)
+
+	for _, arg := range args {
+		arg = replaceEnvVars(arg, envVars) // Replace env variables
+
+		switch {
+		case strings.HasPrefix(arg, "--join="):
+			parsedInput.joinCmd = strings.TrimPrefix(arg, "--join=")
+
+		case strings.HasPrefix(arg, "--port="):
+			num, err := parseInt32(strings.TrimPrefix(arg, "--port="))
+			if err != nil {
+				return fmt.Errorf("invalid --port value: %w", err)
+			}
+			parsedInput.sqlPort = num
+
+		case strings.HasPrefix(arg, "--http-port="):
+			num, err := parseInt32(strings.TrimPrefix(arg, "--http-port="))
+			if err != nil {
+				return fmt.Errorf("invalid --http-port value: %w", err)
+			}
+			parsedInput.httpPort = num
+
+		case strings.HasPrefix(arg, "--insecure"):
+			parsedInput.tlsEnabled = false
+
+		// CockroachDB Enterprise Operator automatically adds "--logs" flag if it is not present. "--logtostderr" just
+		case strings.HasPrefix(arg, "--logtostderr"):
+			continue
+
+		default:
+			if matches := flagRegex.FindStringSubmatch(arg); len(matches) == 3 {
```

Comment:
Does this handle flags that don't take an argument? Or do all flags match the regex?

Reply:
All the flags matches this regex which have `key=value`. This consists of almost all the flags. The flags which doesn't follow this regex expression can not go into the flags map as it expects key value pairs.